diff -Naur onnx-1.20.1.orig/onnx/defs/schema.cc onnx-1.20.1/onnx/defs/schema.cc
--- onnx-1.20.1.orig/onnx/defs/schema.cc	2026-02-10 19:51:33
+++ onnx-1.20.1/onnx/defs/schema.cc	2026-02-10 19:53:06
@@ -24,8 +24,18 @@
 #include "onnx/common/assertions.h"
 #endif
 #include "onnx/defs/parser.h"
+#include "onnx/onnxruntime_fix.h"
 
 namespace ONNX_NAMESPACE {
+
+bool ONNXRuntimeFix::_static_registration_disabled = false;
+bool ONNXRuntimeFix::isStaticRegistrationDisabled() {
+  return _static_registration_disabled;
+}
+
+void ONNXRuntimeFix::disableStaticRegistration() {
+  _static_registration_disabled = true;
+}
 // -1 means ONNX schema hasn't been loaded yet
 // 0 means all versions of ONNX schema have been loaded
 // Other positive integer means the ONNX schemas for the specified version have been loaded
@@ -1738,6 +1748,8 @@
   class SchemasRegisterer {
    public:
     SchemasRegisterer() {
+      // Check if static registration is actually disabled
+      if(ONNXRuntimeFix::isStaticRegistrationDisabled()) return;
       // In debug builds, the number of schema registered in this constructor
       // is compared against the number of calls to schema registration macros.
 #ifndef NDEBUG
diff -Naur onnx-1.20.1.orig/onnx/onnxruntime_fix.h onnx-1.20.1/onnx/onnxruntime_fix.h
--- onnx-1.20.1.orig/onnx/onnxruntime_fix.h	1970-01-01 08:00:00
+++ onnx-1.20.1/onnx/onnxruntime_fix.h	2026-02-10 19:52:43
@@ -0,0 +1,14 @@
+#pragma once
+
+namespace ONNX_NAMESPACE {
+
+class ONNXRuntimeFix {
+public:
+  static bool isStaticRegistrationDisabled();
+  static void disableStaticRegistration();
+
+private:
+  static bool _static_registration_disabled;
+};
+
+} // namespace ONNX_NAMESPACE
